jujugui/static/gui/src/app/components/account/account.js

```diff
@@ -0,0 +1,283 @@
+/*
+This file is part of the Juju GUI, which lets users view and manage Juju
+environments within a graphical interface (https://launchpad.net/juju-gui).
+Copyright (C) 2015 Canonical Ltd.
+
+This program is free software: you can redistribute it and/or modify it under
+the terms of the GNU Affero General Public License version 3, as published by
+the Free Software Foundation.
+
+This program is distributed in the hope that it will be useful, but WITHOUT
+ANY WARRANTY; without even the implied warranties of MERCHANTABILITY,
+SATISFACTORY QUALITY, or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Affero
+General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License along
+with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+'use strict';
+
+YUI.add('account', function() {
+
+  juju.components.Account = React.createClass({
+    listTemplatesXHR: null,
+
+    propTypes: {
+      listTemplates: React.PropTypes.func.isRequired,
+      user: React.PropTypes.object.isRequired,
+      users: React.PropTypes.object.isRequired
+    },
+
+    getInitialState: function() {
+      return {
+        credentials: []
+      };
+    },
+
+    componentWillMount: function() {
+      this.listTemplatesXHR = this.props.listTemplates((error, credentials) => {
+        // XXX This is basic error handling for the initial
+        // implementation. It should be replaced with an error message for the
+        // user in subsequent UI polish work.
+        if (error) {
+          console.error('Unable to list templates', error);
+          return;
+        }
+        this.setState({credentials: credentials});
+      });
+    },
+
+    componentWillUnmount: function() {
+      this.listTemplatesXHR.abort();
+    },
+
+    /**
+      Generate the rows of credentials.
+
+      @method _generateCredentials
+      @returns {Array} The list of credentials.
+    */
+    _generateCredentials: function() {
+      var credentials = this.state.credentials;
+      if (credentials.length === 0) {
```

Comment:
This will make it be a spinner even if the listTemplates call returns no templates. Maybe it should default to this in the template and then when the XHR comes back it triggers a flag to go into a 'none found' state.